apps/meteor/server/startup/migrations/v305.ts

```diff
@@ -0,0 +1,25 @@
+import type { IPermission } from '@rocket.chat/core-typings';
+import { Permissions } from '@rocket.chat/models';
+
+import { upsertPermissions } from '../../../app/authorization/server/functions/upsertPermissions';
+import { addMigration } from '../../lib/migrations';
+
+addMigration({
+	version: 305,
+	name: 'Copy roles from add-team-channel permission to new create-team-channel and crteate-team-group permissions',
+	async up() {
+		// Calling upsertPermissions on purpose so that the new permissions are added before the migration runs
+		await upsertPermissions();
+
+		const addTeamChannelPermission = await Permissions.findOneById<Pick<IPermission, 'roles'>>('add-team-channel', {
```

Comment:
Don't we need to remove this permission after copying? 🤔

Reply:
Actually, we're keeping the `add-team-channel` permission, but making it more specific and clear to users. The client used to apply it for moving existing rooms within teams AND creating rooms inside teams. It'll now only be used for moving existing rooms to teams.
I'll check if we'd like to improve the name of this permission since this is already a breaking change.

Reply:
I decided to replace the `add-team-channel` permission by a new one `move-room-to-team`. The current permission's name can be misleading.